### DISCUSSION TITLE
Help with Auth0 and Google Authentication

### DISCUSSION THREAD

**Discussion by jasonhilldm** [vikejs/vike#1880] [Discussion] [answered]
I am experimenting with using Express, Auth0 and Google Authentication and have used Bati to scaffold an initial app for me. I want to restrict access to certain routes to users of a specific organization (via Google authentication). Ultimately, it would also need RBAC too which can be managed in Auth0.

I have performed the initial Auth0 and Google setup for the test app but I am not sure of the correct way to integrate the authentication and authorization steps into the Vike app. Do I still use the Auth0 React SDK to make calls into Auth0 from within the +guard.ts file?

Is there an example of using Vike + Auth0 + Google authentication that I can refer to?

**Comment by brillout** [comment]
Seems to be more of an Auth0 question than a Vike one. Let us know if you believe there is something about Vike that is blocking you.

**Reply by jasonhilldm** [reply]
Yeah, possibly, but I was hoping for some pointers around where to implement the Auth0 logic within the Vike world, i.e. placing Auth0 code within the +guard.ts files or some higher order component or handler.

The Bati scaffolding for Auth0 with Express implements some auth handlers but unfortunately, it doesn't include any boilerplate code for actually guarding specific pages, so it's not clear what the best practice is for doing that within Vike. Are there any examples that have the page guards implemented as that could be a good starting point?

**Reply by brillout** [reply]
I know many are using Vike with Auth0 but I ain't aware of any public example. Contribution welcome to create one which I'll add to vike.dev.

**Reply by jasonhilldm** [reply]
OK...a quick question then. I am trying to use the Auth0 hook useAuth0() in the +guard.ts file but I get an error "TypeError: Cannot read properties of null (reading 'context')". Is it possible to use hooks within the guard function?

**Reply by phonzammi** [reply, marked as answer]
Please try scaffolding `lucia-auth` or `firebase-auth` with Bati; both include an implementation of guard.ts.

**Reply by jasonhilldm** [reply]
Thanks - that was helpful.
